test/out_dir_in_tests/tests/out_dir_reader.rs

```diff
@@ -0,0 +1,10 @@
+use std::env;
+use std::fs;
+use std::path::PathBuf;
+
+#[test]
+fn can_find_the_out_dir_file() {
+    let out_path = PathBuf::from(env::var("OUT_DIR").unwrap());
```

Comment:
Ditto.

Reply:
Nevertheless, the tests pass under `cargo test`, and I think the behavior should be the same.

Reply:
> For integration tests, I am not sure it makes sense for this env variable to be something only available at compile time.

It is available at compile time when run under Cargo and it is available at compile time when compiling unit tests. Further, the Cargo "spec" is explicit that the only way to get it is through `env!` during compilation and not at runtime of the test. It seems Cargo does also set it at runtime although I think that's accidental and maybe target-specific. 

But the idea that a unit test should be able to do `env!("OUT_DIR")` but an integration test shouldn't is strange. Arguably the only difference between a unit test and an integration test is that integration tests only have access to the public API of the crate.

Note in particular that not all targets have `std:::env` available; `#[no_std]` targets don't, and that's probably why the Cargo rules are the way they are.

Reply:
> Further, the Cargo "spec" is explicit that the only way to get it is through `env!` during compilation

I am not sure the "spec"/docs make a ton of sense here. Previously discussed on slack here: https://app.slack.com/client/TA4K1KQ87/CSV56UT0F/thread/CSV56UT0F-1632336998.082200

The goal is really just to match what Cargo does to a reasonable degree--I'll add some tests that use the macro as well.

> But the idea that a unit test should be able to do `env!("OUT_DIR")` but an integration test shouldn't is strange.

I didn't intend to suggest this. What I meant is that an integration test might reasonably need access to `OUT_DIR` at runtime, in addition to compile time.